klog.go

```diff
@@ -520,6 +532,75 @@ func (l *loggingT) setVState(verbosity Level, filter []modulePat, setFilter bool
 
 var timeNow = time.Now // Stubbed out for testing.
 
+// CaptureState gathers information about all current klog settings and
+// returns a function that, when called, restores those settings.
+// This does not invalidate caches. The restore function may get called
+// more than once.
+func CaptureState() func() {
```

Comment:
Looks like you proposed very compact interface, however I don't find it very clear. Not sure if what is more idiomatic way to express it in Go, however I find:
```
state := klog.CaptureState()
defer state.Restore()
```
much more readable than
```
defer CaptureState()()
```

Reply:
The precedence that I was following is this idiom in Kubernetes:
```
defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.NonPreemptingPriority, true)()
```

I agree that two lines as suggested by you is clearer.

Reply:
I've change the implementation accordingly.

@serathius : can you take another look?